sqlglot/tokens.py

```diff
@@ -451,9 +451,11 @@ class Tokenizer(metaclass=_Tokenizer):
         },
         **{
             f"{prefix}{key}": TokenType.BLOCK_END
-            for key in ("}}", "%}", "#}")
+            for key in ("%}", "#}")
             for prefix in ("", "+", "-")
         },
+        "+}}": TokenType.BLOCK_END,
+        "-}}": TokenType.BLOCK_END,
```

Comment:
Discussed in slack w/ Toby -- Matching a `}}` block end after this PR will require a pair match on `R_BRACE`.